### DISCUSSION TITLE
Why do I need different Z-offsets?

### DISCUSSION THREAD

**Discussion by samuelT2** [mriscoc/Ender3V2S1#830] [Q&A]
Hi

My z-offset is set to -2.10 which does print an almost perfect 0.2mm first layer.
As soon as it comes to the second layer, the nozzle scratches the first layer and I need to manually fine-tune the z-offset to -2.05.

Any hints why this is happening?

Printer: E3V2Neo. I'm running a mesh before every print, my start G-code is below.

(Let me know if discussions should only be for firmware related topics - but I like the community here within your repo.)

`G90 ; use absolute coordinates`
`M83 ; extruder relative mode`
`M104 S150 ; set temporary nozzle temp to prevent oozing during homing and auto bed leveling`
`M140 S[first_layer_bed_temperature] ; set bed temp`
`G4 S30 ; allow partial nozzle warmup`
`; Setup firmware retraction settings`
`M207 F{ retract_speed[0] * 60} S[retract_length_0] Z[retract_lift_0]`
`M208 F{ deretract_speed[0] * 60} S[retract_restart_extra_0]`
`G28 ; home all`
`; M420 S1 Z2; Use a mesh leveling up to 2mm (a valid mesh must be loaded into RAM);`
`G29 P1 ; https://github.com/mriscoc/Ender3V2S1/wiki/Slicer-G-code-Scripts#optional`
`G29 A  ; active the UBL system`
`G1 Z2 F240`
`G1 X2 Y10 F3000`
`M104 S[first_layer_temperature] ; set extruder temp`
`M190 S[first_layer_bed_temperature] ; wait for bed temp`
`M109 S[first_layer_temperature] ; wait for extruder temp`
`G1 Z0.28 F240`
`G92 E0`
`G1 Y190 E15 F1500 ; intro line`
`G1 X2.3 F5000`
`G92 E0`
`G1 Y10 E15 F1200 ; intro line`
`G92 E0`

**Comment by TheWebMachine** [comment]
Perhaps it is the elephant foot compensation in your slicer? This alters amount of filament extruded for first layer to maintain dimensional accuracy. If -2.05 on first layer is too far away, try turning off or reducing EF compensation in your slicer.

**Comment by samuelT2** [comment]
Great hint for something I didn't even know it existed, thank you @TheWebMachine. You're proposing to set this to 0.0 then? I've never touched that setting in PrusaSlicer, it's the default setting for a 0.1mm print.

![image](https://user-images.githubusercontent.com/10983143/228490254-ea0cbc6e-0ee7-45d0-be6d-795fdb224fd3.png)

**Comment by mriscoc** [comment]
Sometimes a Z-hop of 0.2 can help with that.

**Comment by samuelT2** [comment]
Cool, thank you @mriscoc - I'll set this to 0.2mm for my next print and see if it works out.

Edit: Since I use firmware retraction (G10 & G11), I'll change this to 0.2 in the retraction settings of the printer itself, rather then in the slicer.

![image](https://user-images.githubusercontent.com/10983143/228553979-7bc586ee-f574-403e-ab4f-6fb8d51a0e65.png)

**Comment by c4tkybnm** [comment]
@samuelT2 did you end up resolving this? i am having the same issue, first layer is perfect but subsequent layers bubble or are squished unless i raise the z-offset .05mm. Happens with different filaments (all PLA), different temps, speeds, different layer heights (although my first layer is always .2mm), turned off/on elephant's foot compensation, also have z-hop enabled. Once I raise z-offset .05mm after the first layer, then all subsequent layers are perfect.

**Comment by samuelT2** [comment]
@c4tkybnm No, not entirely. I played around with Lift Z and my feeling was, it went a bit better. But the nozzle is still scratching the print from time to time. The "best" solution is still waiting for the first layer to complete and then raise the Z offset by 0.05mm.

Honestly, I'm thinking getting an Original Prusa MK4 kit, once it's available. Creality might have it's advantages - but most of the time I'm busy fixing stuff like the issue above then actual printing. It's a very nice entry printer and it forces you to get into the firmware rabbit hole with all it's possibilities, but yeah, guess I'm done with it at some point.

**Comment by Kaan88** [comment]
This happens when you are overextruding and the surface is higher than first layer height settings. Try tuning your extrusion multiplier.
 I doubt Prusa will be different if it extrudes at the same rate, but it'll definitely be in everything else.

**Comment by samuelT2** [comment]
Thank you @Kaan88 - I'll give this one a shot as well.

I know, but I'm hoping that the presets provided in the Prusa Slicer for Prusa printers will work more precisely with less finetuning needed. Don't get me wrong, the deep dives into all the possible settings are fun, but sometimes I just wish that things would work a bit better out of the box.

**Comment by c4tkybnm** [comment]
thanks @Kaan88, your post gave me an idea as i remember this issue started happening around the time i switched to a metal extruder meaning i also replaced the extruder gear. i ended up reusing the extruder gear from the stock plastic extruder and it seems to have fixed it. 

question though, wouldn't tuning the extrusion multiplier affect all layers and not just the first?

**Reply by Kaan88** [reply]
Yes and if you underextrude too much the part will be weaker and very slightly smaller. The aim is to not overextrude and find a good balance. 
By the way, some slicers (orca, superslicer) have top and bottom only extrusion multiplier too. Sometimes I use these to get a nice top finish.

**Reply by c4tkybnm** [reply]
Aah I just downloaded orca, good to know you can set extrusion for initial layers. Do you have any good orca profiles for e3v2? Or are default good enough?

**Reply by Kaan88** [reply]
I started with that but copied settings from prusa for S1, specificaly line width stuff. Slighty thicker lines helpes with overhangs.

**Reply by c4tkybnm** [reply]
okay interesting. i seem to get better print results with cura to be honest. possibly orca is better suited for bambu printers which makes sense

**Reply by dfinberg** [reply]
cura also has a top/bottom flow parameter (search flow), but not one for each.  There's also an initial layer flow setting.

If you've played with your extruder you should retune both the esteps, as well as the flow.

**Reply by c4tkybnm** [reply]
thanks @dfinberg have sorted this after calibrating esteps and flow correctly. perfect layers and parts fit together perfectly now too. 

CC @samuelT2 just in case you haven't been able to resolve yet
